ui/common/src/socket.ts

```diff
@@ -90,49 +93,44 @@ export default class StrongSocket {
         ...(settings.params || {}),
       },
     };
-    const customPingDelay = storedIntProp('socket.ping.interval', 2500)();
-
-    this.options = {
-      ...StrongSocket.defaultOptions,
-      ...(settings.options || {}),
-      pingDelay: customPingDelay > 400 ? customPingDelay : 2500,
-    };
     this.version = version;
     site.pubsub.on('socket.send', this.send);
     this.connect();
+    this.flushStats();
+    window.addEventListener('pagehide', () => this.storeStats({ event: 'pagehide' }));
   }
 
   sign = (s: string): void => {
     this._sign = s;
     this.ackable.sign(s);
   };
 
-  connect = (): void => {
+  private connect = (): void => {
     this.destroy();
     if (!isOnline()) {
       document.body.classList.remove('online');
       document.body.classList.add('offline');
       $('#network-status').text(site ? site.trans('noNetwork') : 'Offline');
-      this.scheduleConnect(1000);
+      this.scheduleConnect(4000);
       return;
     }
-    this.autoReconnect = true;
-    const fullUrl = xhr.url(this.options.protocol + '//' + this.baseUrl() + this.url, {
+    this.lastUrl = xhr.url(this.options.protocol + '//' + this.nextBaseUrl() + this.url, {
       ...this.settings.params,
       v: this.version === false ? undefined : this.version,
     });
-    this.debug('connection attempt to ' + fullUrl);
+    this.debug('connection attempt to ' + this.lastUrl);
     try {
-      const ws = (this.ws = new WebSocket(fullUrl));
+      const ws = (this.ws = new WebSocket(this.lastUrl));
       ws.onerror = e => this.onError(e);
-      ws.onclose = e => this.onClose(e, fullUrl);
+      ws.onclose = this.onClose;
       ws.onopen = () => {
-        this.debug('connected to ' + fullUrl);
-        this.onSuccess();
+        this.lastUrl = ws.url;
+        this.debug('connected to ' + this.lastUrl);
         const cl = document.body.classList;
+        if (site.pubsub.past('socket.hasConnected')) cl.add('reconnected');
         cl.remove('offline');
         cl.add('online');
-        cl.toggle('reconnected', this.nbConnects > 1);
+        this.onSuccess();
```

Comment:
some non-critical stuff shuffled around here, we could actually move all of it to onSuccess, or onOpen if we wanted to maintain uniformity with the websocket event